src/main/java/io/confluent/connect/hdfs/wal/FSWAL.java

```diff
@@ -183,13 +189,166 @@ private void commitEntriesToStorage(Map<WALEntry, WALEntry> entries) {
     }
   }
 
+  /**
+   * Extract the latest offset and file path from the WAL file.
+   * Attempt with the most recent WAL file, and fall back to the old file if it's not applicable.
+   *
+   * <p> The old WAL is used when the most recent WAL file has already been truncated
+   * and is not existent, which may happen when the connector has flushed all records,
+   * and is restarted. </p>
+   *
+   * @return the latest offset and filepath from the WAL file or null
+   */
+  @Override
+  public FilePathOffset extractLatestOffset() {
+    String oldWALFile = logFile + TRUNCATED_LOG_EXTENSION;
+    try {
+      FilePathOffset latestOffset = null;
+      if (storage.exists(logFile)) {
+        log.trace("Restoring offset from WAL file: {}", logFile);
+        if (reader == null) {
+          reader = newWalFileReader();
+        } else {
+          // reset read position after apply()
+          reader.seekToFirstRecord();
+        }
+        List<String> committedFileBatch = getLastFilledBlockFromWAL(reader);
+        // At this point the committedFilenames list will contain the
+        // filenames in the last BEGIN-END block of the file. Find the latest offsets among these.
+        latestOffset = getLatestOffsetFromList(committedFileBatch);
```

Comment:
```suggestion
        return getLatestOffsetFromList(committedFileBatch);
```
then we can remove the null check in the if statement

Reply:
Here it would still be necessary to check on the result of this function. `getLatestOffsetFromList()` returns `null` when no offset is found. In that case we would need to move up the null check because we don't want to return `null` without checking the old file. In that case we may keep the current logic?